tests/src/tests/component_test_utils.go

```diff
@@ -0,0 +1,228 @@
+	}
+
+	generateComponent(&components[index], componentType)
+	devfile.SchemaDevFile.Components = components
+
+	return components[index].Name
+
+}
+
+func generateComponent(component *schema.Component, componentType schema.ComponentType) {
+
+	component.Name = GetRandomUniqueString(8, true)
+	LogMessage(fmt.Sprintf("   ....... Name: %s", component.Name))
+
+	if componentType == schema.ContainerComponentType {
+		component.Container = createContainerComponent()
+	} else if componentType == schema.VolumeComponentType {
+		component.Volume = createVolumeComponent()
+	}
+}
+
+func createContainerComponent() *schema.ContainerComponent {
+
+	LogMessage("Create a container component :")
+
+	containerComponent := schema.ContainerComponent{}
+	setContainerComponentValues(&containerComponent)
+
+	return &containerComponent
+
+}
+
+func createVolumeComponent() *schema.VolumeComponent {
+
+	LogMessage("Create a volume component :")
+
+	volumeComponent := schema.VolumeComponent{}
+	setVolumeComponentValues(&volumeComponent)
+
+	return &volumeComponent
+
+}
+
+func setContainerComponentValues(containerComponent *schema.ContainerComponent) {
+
+	containerComponent.Image = GetRandomUniqueString(8+GetRandomNumber(10), false)
+
+	if GetBinaryDecision() {
+		numCommands := GetRandomNumber(3)
+		containerComponent.Command = make([]string, numCommands)
+		for i := 0; i < numCommands; i++ {
+			containerComponent.Command[i] = GetRandomString(4+GetRandomNumber(10), false)
+			LogMessage(fmt.Sprintf("   ....... command %d of %d : %s", i, numCommands, containerComponent.Command[i]))
+		}
+	}
+
+	if GetBinaryDecision() {
+		numArgs := GetRandomNumber(3)
+		containerComponent.Args = make([]string, numArgs)
+		for i := 0; i < numArgs; i++ {
+			containerComponent.Args[i] = GetRandomString(8+GetRandomNumber(10), false)
+			LogMessage(fmt.Sprintf("   ....... arg %d of %d : %s", i, numArgs, containerComponent.Args[i]))
+		}
+	}
+
+	containerComponent.DedicatedPod = GetBinaryDecision()
+	LogMessage(fmt.Sprintf("   ....... DedicatedPod: %t", containerComponent.DedicatedPod))
+
+	if GetBinaryDecision() {
+		containerComponent.MemoryLimit = strconv.Itoa(4+GetRandomNumber(124)) + "M"
+		LogMessage(fmt.Sprintf("   ....... MemoryLimit: %s", containerComponent.MemoryLimit))
+	}
+
+	if GetBinaryDecision() {
+		setMountSources := GetBinaryDecision()
+		containerComponent.MountSources = &setMountSources
+		LogMessage(fmt.Sprintf("   ....... MountSources: %t", *containerComponent.MountSources))
+
+		if setMountSources {
+			containerComponent.SourceMapping = "/" + GetRandomString(8, false)
+			LogMessage(fmt.Sprintf("   ....... SourceMapping: %s", containerComponent.SourceMapping))
+		}
+	}
+
+	if GetBinaryDecision() {
+		containerComponent.Env = AddEnv(GetRandomNumber(4))
+	} else {
+		containerComponent.Env = nil
+	}
+
+	if GetBinaryDecision() {
+		containerComponent.VolumeMounts = AddVolume(GetRandomNumber(4))
+	} else {
+		containerComponent.Env = nil
```

Comment:
should be `VolumeMounts`?

Reply:
if it's `nil`, you can just leave it empty instead of setting it as nil. same as above

Reply:
yes, oops